scheduler/setup_funcs.go

```diff
@@ -35,6 +36,31 @@ func cacheTaskGroups(comparator *CmpBasedTaskComparator) error {
 	return nil
 }
 
+func backfillTaskGroups(comparator *CmpBasedTaskComparator) error {
+	catcher := grip.NewBasicCatcher()
+
+OUTERLOOP:
+	for _, t := range comparator.tasks {
```

Comment:
This means that we will still parse the yaml of every group though. We should probably fold `cacheTaskGroups` into this function. It also will likely change when the parseProject work goes in.

Reply:
yes, but we only process each task once, because task group orders are for now, 1 indexed, and there may always be new tasks, so I think its safe... 

cacheTaskGroups iterates over versions, so I think they're ~orthogonal